File: ws/lainez/u2/ws21basicOperation/BasicOperations/src/test/java/ec/edu/espe/basicoperations/controller/PersonControllerTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */
package ec.edu.espe.basicoperations.controller;

import java.time.LocalDate;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author WINDOWS
 */
public class PersonControllerTest {
    
    public PersonControllerTest() {
    }

    /**
     * Test of computeAgeInDays method, of class PersonController.
     */
    @org.junit.jupiter.api.Test
    public void testComputeAgeInDays() {
        System.out.println("computeAgeInDays");
        LocalDate birthDate = LocalDate.parse("2005-03-06");
        PersonController instance = new PersonController();
        int expResult = 7064;
        int result = instance.computeAgeInDays(birthDate);
        assertEquals(expResult, result);
        
    }

    /**
     * Test of computeAgeInYears method, of class PersonController.
     */
    @org.junit.jupiter.api.Test
    public void testComputeAgeInYears() {
        System.out.println("computeAgeInYears");
        LocalDate birthDate = LocalDate.parse("2005-03-06");
        PersonController instance = new PersonController();
        int expResult = 19;
        int result = instance.computeAgeInYears(birthDate);
        assertEquals(expResult, result);
        
    }
    
}

File: hw/raura/U2/hw13ToOurProgrammingLanguage/src/test/PersonController.test.js
const PersonController = require('../controller/PersonController')

test('computeAgeInDays', () => {
    console.log('computeAgeInDays');
    const birthDate = new Date('2004-04-21');
    const instance = new PersonController();
    const expResult = 7388;
    const result = instance.computeAgeInDays(birthDate);
    expect(result).toBe(expResult);
});

test('computeAgeInYears', () => {
    console.log('computeAgeInYears');
    const birthDate = new Date('2004-04-21');
    const instance = new PersonController();
    const expResult = 20;
    const result = instance.computeAgeInYears(birthDate);
    expect(result).toBe(expResult);
});
